compile.js

```diff
@@ -415,3 +462,72 @@ function getDefaultWriteTest(ctx, field) {
 function isPacked(field) {
     return field.options.packed === 'true';
 }
+
+function getTsType(field) {
+    let type = field.type;
+
+    if (fieldShouldUseStringAsNumber(field)) type = 'string';
+    else if (fieldTypeIsNumber(field)) type = 'number';
+    else if (field.type === 'bytes') type = 'Uint8Array';
+    else if (field.type === 'bool') type = 'boolean';
+
+    return field.repeated ? `Array<${type}>` : type;
+}
+
+function getMapTsType(fields) {
+    const key = getTsType(fields[0]);
+    const value = getTsType(fields[1]);
+    return {key, value};
+}
+
+/**
+ * @param {string} type
+ * @param {string} name
+ * @param {{name: string; type: string; required: boolean}} [fields]
+ * @returns {string}
+ */
```

Comment:
Nit: should we add JSDoc to some of the functions here if the file isn't typechecked and the rest of the code doesn't have it?

Reply:
Yep, will drop JSDoc here for consistency.